internal/controller/k0smotron.io/k0smotroncluster_controller.go

```diff
@@ -246,6 +259,71 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	return ctrl.Result{}, nil
 }
 
+func (r *ClusterReconciler) ensureServiceAccountRoleAndBinding(ctx context.Context, kmc *km.Cluster) error {
+	sa := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceAccountName,
+			Namespace: kmc.Namespace,
+		},
+	}
+	if err := r.Get(ctx, client.ObjectKey{Name: serviceAccountName, Namespace: kmc.Namespace}, sa); client.IgnoreNotFound(err) != nil {
+		return err
+	} else if err != nil {
+		if err := r.Create(ctx, sa); err != nil {
+			return err
+		}
+	}
```

Comment:
While this pattern is good in general, I think this creates a bit of a problem. Say in some future version we need to change the roles/bindings, we would not be able to do so since the old ones would already exist